core/include/drivers/gic.h

```diff
@@ -56,4 +56,74 @@ void gic_cpu_init(struct gic_data *gd);
 void gic_it_handle(struct gic_data *gd);
 
 void gic_dump_state(struct gic_data *gd);
+
+
+#if defined(CFG_ARM_GICV3)
+#if defined(ARM64)
+static inline uint32_t read_icc_ctlr_el1(void)
```

Comment:
These inline functions belongs in `core/arch/arm/include/arm64.h`.
You can guard them with `#if defined(CFG_ARM_GICV3)` if you like.